### DISCUSSION TITLE
write output at given particle age

### DISCUSSION THREAD

**Discussion by JamiePringle** [Parcels-code/Parcels#1282] [Q&A]
Currently, ocean parcels writes out all particle positions at a given time. I would like to write out particles positions at a given age since release (the particles are released at a higher frequency than the output frequency). Is there an easy way to do this?

I am currently thinking of a hack-ish way to do this -- keeping position at last age output time as separate variables (last_x,last_y,last_z) which I update at the right age. I then save those variables. 

Or I could write out at a high-frequency, and then post-process the files. But that can lead to very large initial output files. 

But is there a more efficient way?

Thanks, Jamie

**Comment by erikvansebille** [comment]
Hmm, if you only want to write at one specific age then you could use the `write_on_delete=True` option in [ParticleFile](https://oceanparcels.org/gh-pages/html/#module-parcels.particlefile.baseparticlefile), in combination with a kernel that deletes a particle when it reaches a certain age

Alternatively, you could do a separate run for each starttime (or combine start times that are multiples of the age at which to write), so that you don't mix them. But that of course is more heavy on I/O for the hydrodynamic fields (because each run will have to read in all the hydrodynamic data).

**Comment by erikvansebille** [comment]
More fundamentally, there's currently no option to write at a specific age, although that should not be too difficult to implement if someone wants to have a go at it. The key line to change is 
https://github.com/OceanParcels/parcels/blob/7acb4444caf1dec8b7e9a40a5510654c390877db/parcels/particlefile/baseparticlefile.py#L222
and the associated function
https://github.com/OceanParcels/parcels/blob/7acb4444caf1dec8b7e9a40a5510654c390877db/parcels/collection/collectionsoa.py#L801-L809

where now all particles are gathered whose `time` attribute is close to the writing time. That should somehow be changed to `age`. API-wise, I guess it would be nice if users could provide a list of ages that particles should be written at
